refinery/core/models.py

```diff
@@ -552,31 +552,17 @@ def delete(self, **kwargs):
         pre_isa_archive associated with the DataSet if one exists.
         """
 
-        try:
-            self.get_isa_archive().delete()
-        except AttributeError as e:
-            logger.debug("DataSet has no isa_archive to delete: %s", e)
-
-        try:
-            self.get_pre_isa_archive().delete()
-        except AttributeError as e:
-            logger.debug("DataSet has no pre_isa_archive to delete: %s", e)
-
-        related_investigation_links = self.get_investigation_links()
-
-        for investigation_link in related_investigation_links:
-            node_collection = investigation_link.get_node_collection()
-            try:
-                node_collection.delete()
-            except Exception as e:
-                logger.error("Couldn't delete NodeCollection: %s", e)
-
         # terminate any running file import tasks
         for file_store_item in self.get_file_store_items():
             file_store_item.terminate_file_import_task()
 
+        related_investigation_links = self.get_investigation_links()
+
         try:
-            super(DataSet, self).delete()
+            with transaction.atomic():
+                for investigation_link in related_investigation_links:
+                    investigation_link.get_node_collection().delete()
+                super(DataSet, self).delete()
```

Comment:
Yes I believe it should stay. Imagine if someone extended parent's `.delete()` and that code now could cause a faulty child deletion

Reply:
This could cause hard to diagnose errors if exception handling is added to delete() in parent. "Avoid catching exceptions inside atomic" https://docs.djangoproject.com/en/1.8/topics/db/transactions/#controlling-transactions-explicitly